dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceOutputStream.java

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+ * HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *******************************************************************************/
+package org.hpccsystems.dfs.client;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.io.OutputStream;
+
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+import org.json.JSONObject;
+import org.apache.log4j.Logger;
+import org.hpccsystems.commons.ecl.RecordDefinitionTranslator;
+import org.hpccsystems.commons.ecl.FieldDef;
+
+public class RowServiceOutputStream extends OutputStream
+{
+    private static final Logger      log                           = Logger.getLogger(RowServiceOutputStream.class.getName());
+    private static int DEFAULT_CONNECT_TIMEOUT_MILIS = 1000;       // 1 second connection timeout
```

Comment:
if it takes us longer than 1sec to connect, we might have bigger issues, but it might be worthwhile to provide a mechanism for the caller to overwrite the 1sec default ( does Spark client have connection timeouts we need to respect?)